cpp/foxglove/tests/test_mcap.cpp

```diff
@@ -0,0 +1,147 @@
+#include <foxglove/channel.hpp>
+#include <foxglove/mcap.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_string.hpp>
+
+#include <filesystem>
+#include <fstream>
+using Catch::Matchers::ContainsSubstring;
+using Catch::Matchers::Equals;
+
+class FileCleanup {
+public:
+  FileCleanup(const std::string& path)
+      : _path(path) {}
+  ~FileCleanup() {
+    if (std::filesystem::exists(_path)) {
+      std::filesystem::remove(_path);
+    }
+  }
+
+private:
+  std::string _path;
+};
+
+TEST_CASE("Open new file and close mcap writer") {
+  FileCleanup cleanup("test.mcap");
+
+  foxglove::McapWriterOptions options = {};
+  options.path = "test.mcap";
+  options.create = true;
+  foxglove::McapWriter writer(options);
+  writer.close();
+
+  // Check if test.mcap file exists
+  REQUIRE(std::filesystem::exists("test.mcap"));
```

Comment:
at some point we may want an e2e test that actually reads the file contents – for now, maybe it would be good to assert what the file size is? (just to prove that something was actually written)  But on the other hand that would lead to test failures if we ever change some details of the writer, like the default library string, so feel free to say no :)

Reply:
maybe that the file size is bigger than X would be less fragile

I tried adding our mcap c++ library as a test dependency, but that road quickly became painful. At some point we should probably do it though. Cargo is so, so nice.